### ISSUE TITLE
segfault/memory error running pyxel apps with python 3.9.10 on ubuntu 21.04

### ISSUE THREAD

**Issue by backwardspy** [kitao/pyxel#307]
**describe the bug**
when running any pyxel apps on my ubuntu 21.04 installation with python 3.9.10, there is a memory error when pyxel calls `update` inside `pyxel.run(update, draw)`.

**to reproduce**
run any pyxel application, for example:

```python
import pyxel

def update():
    print("updating")

def draw():
    print("drawing")

pyxel.init(200, 200)
pyxel.run(update, draw)
```

results in:

```shell
$ python repro.py
fish: Job 1, 'python repro.py' terminated by signal SIGSEGV (Address boundary error)
```

this happens with every other pyxel application i've tried, including `pyxeleditor`.

**expected behavior**
the program should run with no memory error

**desktop**
 - OS: ubuntu 21.04
 - versions: python 3.9.2, pyxel 1.4.3

**additional context**
i tried the same code on python 3.8.10, and got the same result.

i also built a custom `libpyxelcore.so` with debug symbols enabled and ran the above `repro.py` with python loaded into gdb. the crash occurs on line [203 of `pyxelcore/system.cc`](https://github.com/kitao/pyxel/blob/master/pyxel/core/src/pyxelcore/system.cc#L203), when pyxel tries to call `update()`.
the function pointers for `update` and `draw` are not null:

```
update: 0x7ffff72ac010
draw: 0x7ffff72ac048
```

this is as far as i was able to debug by myself, i wasn't sure where to go from here.

**Comment by backwardspy** [comment]
i have made a small discovery; this crash only happens with my `pyenv` python versions. if i run the game using a python installed with `homebrew` (linuxbrew) it does not crash. i haven't yet been able to figure out why that is.

**Comment by kitao** [comment]
The core engine is re-implemented in 1.5.0. Please check it again.

**Comment by merwok** [comment]
The problem I mentioned still exists.
The shared lib should be built from source, not included in the repository.
Platform-specific wheels should be built (windows, macos, manylinux, musllinux) and distributed.

I will open a ticket for this and work on PRs in the second half of the month.
